### PR TITLE
feat: Static validation rules

### PR DESCRIPTION
**Pull request by alex-clayton-1**
## Description

Implemented rules for static validation.

## Context

[DTOSS-3159](https://nhsd-jira.digital.nhs.uk/browse/DTOSS-3159)

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: application/CohortManager/src/Functions/Shared/Data/Database/CreateDemographicData.cs [hunk 1]
```diff
@@ -185,14 +185,14 @@ private Demographic GetDemographic(IDbCommand command)
                 demographic.OtherGivenNames = reader["OTHER_GIVEN_NAME"] == DBNull.Value ? null : reader["OTHER_GIVEN_NAME"].ToString();
                 demographic.Surname = reader["FAMILY_NAME"] == DBNull.Value ? null : reader["FAMILY_NAME"].ToString();
                 demographic.PreviousSurname = reader["PREVIOUS_FAMILY_NAME"] == DBNull.Value ? null : reader["PREVIOUS_FAMILY_NAME"].ToString();
-                demographic.DateOfBirth = reader["DATE_OF_BIRTH"] == DBNull.Value ? null : reader["DATE_OF_BIRTH"].ToString();
+                demographic.DateOfBirth = reader["DATE_OF_BIRTH"] == DBNull.Value ? null : DateTime.Parse(reader["DATE_OF_BIRTH"].ToString()).ToString("yyyyMMdd");
```
Comment (stephhou) [inline application/CohortManager/src/Functions/Shared/Data/Database/CreateDemographicData.cs:188]: What will the result be if this parsing if the DoB is just a year (ccyy) or a year and month ccyyMM?

Comment (alex-clayton-1) [reply application/CohortManager/src/Functions/Shared/Data/Database/CreateDemographicData.cs:188]: The DATE_OF_BIRTH column is of type DATE so I think it will always be in the SQL DATE format (YYYY-MM-DD). What I'm doing here is converting it to one of the formats that static validation expects. The DoB validation rule can also validate dates that are ccyy or ccyymm.

File: application/CohortManager/src/Functions/CaasIntegration/processCaasFile/processCaasFile.cs [hunk 2]
```diff
@@ -53,14 +53,14 @@ public async Task<HttpResponseData> Run([HttpTrigger(AuthorizationLevel.Anonymou
...
             };
-            var json = JsonSerializer.Serialize(basicParticipantCsvRecord);
 
             switch (participant.RecordType?.Trim())
             {
                 case Actions.New:
                     add++;
                     try
                     {
+                        var json = JsonSerializer.Serialize(basicParticipantCsvRecord);
```
Comment (SamRobinson75684) [inline application/CohortManager/src/Functions/CaasIntegration/processCaasFile/processCaasFile.cs:63]: why are these changes needed?

Comment (alex-clayton-1) [reply application/CohortManager/src/Functions/CaasIntegration/processCaasFile/processCaasFile.cs:63]: Not all cases in the switch statement will use the same json in their HTTP requests. Add, update and delete use basicParticipantCsvRecord whereas the error case uses ParticipantCsvRecord.

**Review by SamRobinson75684 — APPROVED** [review]

**Review by andrewmorris43 — APPROVED** [review]
All looks good to me.
